sched/semaphore/sem_holder.c

```diff
@@ -1035,18 +1014,57 @@ void nxsem_release_holder(FAR sem_t *sem)
 {
   FAR struct tcb_s *rtcb = this_task();
   FAR struct semholder_s *pholder;
+  FAR struct semholder_s *candidate = NULL;
+  unsigned int total = 0;
 
   /* Find the container for this holder */
 
-  pholder = nxsem_findholder(sem, rtcb);
-  if (pholder != NULL && pholder->counts > 0)
+#if CONFIG_SEM_PREALLOCHOLDERS > 0
+  for (pholder = sem->hhead; pholder; pholder = pholder->flink)
+#else
+  int i;
+
+  /* We have two hard-allocated holder structures in sem_t */
+
+  for (i = 0; i < 2; i++)
```

Comment:
```suggestion
  for (i = 0, pholder = &sem->holder[i];
      i < 2 && pholder->counts > 0;
      i++)
```

Reply:
Holder[1]'s count is not necessarily 0 when holder[0]'s count is 0.

Reply:
Yes. My bad. I just thought that we can eliminate `continue`

Reply:
Why `pholder->counts <= 0` with `continue` is checked only in case if number if preallocated holders is 0?

Reply:
If the holder's counts == 0, it will be removed from the list later in nxsem_restore_baseprio, so all holder's counts must > 0 when enter nxsem_release_holder.